framework/db/src/main/java/com/cloud/utils/db/GenericDaoBase.java

```diff
@@ -2055,7 +2046,17 @@ public Integer getCount(SearchCriteria<T> sc) {
 
     @DB()
     protected StringBuilder createCountSelect(SearchCriteria<?> sc, final boolean whereClause) {
-        StringBuilder sql = new StringBuilder(_count);
+        if (sc == null)
+            return null;
+
+        StringBuilder sql;
```

Comment:
Alternatively, you can wrap all the new groupBy changes under `if (sc != null) { ... }`